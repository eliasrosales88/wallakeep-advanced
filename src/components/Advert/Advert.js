import React, { Fragment } from 'react';
import { withRouter } from "react-router-dom";


const Advert = ( props ) => {
  const {photo, name, description, price, id, history, type, tags=[], buttonsActive = true} = props;
  
  const detailHandler = () => {
    history.push("advert/"+ id)
  }
  
  const editHandler = () => {
    history.push("advert/edit/" + id )
    
  }

  
  return (
    <Fragment>
      <div className="card">
        <img src={ photo } className="card-img-top" alt={ name } />
        <div className="card-body">
          <h5 className="card-title">{ name }</h5>
          <p className="card-text"><b>Precio: { price }$</b></p>
          <p className="card-text">{ description }</p>
          <p>
            {type === "sell" ? <span className="badge badge-danger p-2">{type}</span> : <span className="badge badge-success p-2">{type}</span>}
          </p>
          {tags.map((tag, index) =>{
            return (
              <Fragment key={index}>
                <span className="badge badge-primary ml-1">{tag}</span>
              </Fragment>
            )
            })
          }

          {buttonsActive &&
            <div className="mt-1">
              <button onClick={detailHandler} className="btn btn-primary">More</button>
              <button onClick={editHandler} className="btn btn-warning ml-1">Edit</button>
            </div>  
          }
        </div>
      </div>
    </Fragment>
  )
}

export default withRouter(Advert);
