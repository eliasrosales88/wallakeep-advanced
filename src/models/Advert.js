/**
 * Modelo de anuncio en nodepop
 */
export default class Advert {
  /**
   * Constructor
   * @param {Object} Advert 
   */    
  constructor(ad, apiUrl) {
      this._id = ad._id;
      this.createdAt = ad.createdAt;
      this.name = ad.name;
      this.description = ad.description;
      this.price = ad.price;
      this.type = ad.type;
      this.photo = ad.photo.startsWith('/images/')?`${apiUrl.replace('apiv1','')}${ad.photo}`:ad.photo;
      this.tags = ad.tags;
  }

  /**
   * Comprueba si un objeto advert es vÃ¡lido. (Campos obligatorios completos)
   */
  isValid() {
      return  this.name && 
              this.description && 
              this.price > 0 && 
              this.type && 
              this.photo && 
              this.tags && 
              this.tags.length >= 1;
  }
}