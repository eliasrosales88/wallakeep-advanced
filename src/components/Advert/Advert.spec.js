import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Advert from './Advert';


configure({ adapter: new Adapter() });

describe('Advert', () => {
  const defaultProps = {
    key: 'advert1',
    name: 'A name',
    price: 3999.99,
    type: 'buy',
    tags: ["lifestyle", "motor"],
    description: "A description",
    id: 'advert1',
    photo: 'string',
  };
  
  const render = props => shallow(<Advert.WrappedComponent
    key={defaultProps.id}
    name={defaultProps.name}
    price={defaultProps.price}
    type={defaultProps.type}
    tags={defaultProps.tags}
    description={defaultProps.description}
    id={defaultProps.id}
    photo={defaultProps.photo}
    history={{location:{pathname: "/advert/"}}}
     />);
  let wrapper;

  beforeEach(() => {
    wrapper = render();
  });


  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 2 tags', () => {
      
    expect(wrapper.find('.badge-primary')).toHaveLength(2);
  });

});
