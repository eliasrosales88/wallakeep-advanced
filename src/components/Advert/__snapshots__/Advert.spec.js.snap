// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advert should render 1`] = `
<Fragment>
  <div
    className="card"
  >
    <img
      alt="A name"
      className="card-img-top"
      src="string"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        A name
      </h5>
      <p
        className="card-text"
      >
        <b>
          Precio: 
          3999.99
          $
        </b>
      </p>
      <p
        className="card-text"
      >
        A description
      </p>
      <p>
        <span
          className="badge badge-success p-2"
        >
          buy
        </span>
      </p>
      <span
        className="badge badge-primary ml-1"
      >
        lifestyle
      </span>
      <span
        className="badge badge-primary ml-1"
      >
        motor
      </span>
      <hr />
      <br />
    </div>
  </div>
</Fragment>
`;
